[CCR] Clarify that version ordering includes the patch version (#7529)
elastic/docs-content: deploy-manage/tools/cross-cluster-replication.md

* In a uni-directional configuration, one cluster contains only leader indices, and the other cluster contains only follower indices.
* In a bi-directional configuration, each cluster contains both leader and follower indices.

> In a uni-directional configuration, the cluster containing follower indices (the *local* cluster) must be compatible with the cluster containing leader indices (the *remote* cluster). Both of the following conditions must be met:

Reviewer: ```suggestion
## Version compatibility

In a uni-directional configuration, the cluster containing follower indices (the *local* cluster) must be compatible with the cluster containing leader indices (the *remote* cluster). Both of the following conditions must be met:
```

Considering the compatibility statements are growing, I'd suggest to create its own section.

@yetanothertw , what do you think of this PR?

Currently the preview would render like (showing the entire introduction):

<img width="701" height="1009" alt="image" src="https://github.com/user-attachments/assets/4f46b4d2-39d1-4c05-a3df-11948ba14d9f" />

I think this PR adds value but probably we need to put the new content under its own H2 section.

Reviewer 2: That makes sense to me, @eedugon

> * The local cluster must be running **the same or newer** version of {{es}} as the remote cluster. This comparison includes the patch version: if the remote cluster is running `X.Y.3`, the local cluster must be running `X.Y.3` or higher. `X.Y.1` is not sufficient, even though both clusters are on the same minor version.

Reviewer: ```suggestion
* The local cluster must be running **the same or newer** version of {{es}} as the remote cluster. This comparison includes the patch version: for example, if the remote cluster is running `9.4.3`, the local cluster must be running `9.4.3` or higher. `9.4.1` is not sufficient, even though both clusters are on the same minor version.
```

Reviewer 2: ```suggestion
* The local cluster must be running **the same or newer (later)** version of {{es}} as the remote cluster, including the patch version. For example, a remote cluster running on `9.4.3`, requires the local cluster to run on `9.4.3` or later — `9.4.1` is not sufficient.
```

> * If the local cluster is running a newer version, the versions must also be compatible as outlined in the following matrix.

Reviewer: ```suggestion
* If the local cluster is running a later version than the remote cluster, the versions must also be compatible as outlined in the following matrix.
```

::::{important}
The following matrix groups versions into ranges, and expresses only the second condition. It does not relax the first: within a single cell of the matrix, the local cluster must still be running the same or a higher patch version than the remote cluster.
::::

:::::{dropdown} Version compatibility matrix

> :name: ccr-version-compatibility

Reviewer: Not in the scope of this PR, but @yetanothertw , don't you think the big table is outdated? It doesn't show 9.x versions. Maybe we should create a separate issue to address that (if it's confirmed)

Author: I agree with that.

Reviewer 2: Sure, Edu, let's create another issue to track that.

* In a uni-directional configuration, one cluster contains only leader indices, and the other cluster contains only follower indices.
* In a bi-directional configuration, each cluster contains both leader and follower indices.

## Version compatibility

In a uni-directional configuration, the cluster containing follower indices (the *local* cluster) must be compatible with the cluster containing leader indices (the *remote* cluster). Both of the following conditions must be met:

> * The local cluster must be running **the same or newer** version of {{es}} as the remote cluster. This comparison includes the patch version: for example, if the remote cluster is running `9.4.3`, the local cluster must be running `9.4.3` or higher. `9.4.1` is not sufficient, even though both clusters are on the same minor version.

Reviewer 2: ```suggestion
* The local cluster must be running **the same or newer (later)** version of {{es}} as the remote cluster, including the patch version. For example, a remote cluster running on `9.4.3`, requires the local cluster to run on `9.4.3` or later — `9.4.1` is not sufficient.
```

> * If the local cluster is running a newer version than the remote cluster, the versions must also be compatible as outlined in the following matrix.

Reviewer 2: ```suggestion
* If the local cluster is running a later version than the remote cluster, the versions must also be compatible as outlined in the following matrix.
```

::::{important}

> The following matrix groups versions into ranges, and expresses only the second condition. It does not relax the first: within a single cell of the matrix, the local cluster must still be running the same or a higher patch version than the remote cluster.

Reviewer 2: I recommend rephrasing to clarify the conditions we're referring to (rather than just saying the second condition or the first). 

```suggestion
The following matrix describes the version compatibility between local and remote clusters when the local cluster is running a later version than the remote. The patch version requirement still applies: within any cell, the local cluster must still match or exceed the remote cluster's exact patch version.
```
